### PR TITLE
Feature/oauth

### PR DESCRIPTION
**Pull request by lzhang433**
implemented oauth protections for whole site + login page (to be improved)

### CODE REVIEW

**Comment by christianMosquera** [comment]
Make sure that the database.yml file has this line under the production db: url: <%= ENV['DATABASE_URL'] %>

**Review by JoshHare — COMMENTED** [review]
Coding is style is good!!!

File: app/controllers/test2_controller.rb
Comment (christianMosquera) [inline app/controllers/test2_controller.rb:1]: what is this test2_controller?

File: app/controllers/test_controller.rb
Comment (christianMosquera) [inline app/controllers/test_controller.rb:1]: what is this test_controller?

**Review by christianMosquera — COMMENTED** [review]
look at the test_controller stuff, if not needed, then it should be deleted

**Pull request merged**